src/main/java/com/lgtm/easymoney/controllers/UserController.java

```diff
@@ -1,67 +1,93 @@
-    private final AccountRepository accountRepository;
-
     @Autowired
-    public UserController(UserRepository userRepository, AccountRepository accountRepository) {
-        this.userRepository = userRepository;
-        this.accountRepository = accountRepository;
-    }
+    private UserService userService;
+    @PutMapping("/deposit")
+    public ResponseEntity<?> deposit(@Valid @RequestBody BalanceReq req) {
+        // get params
+        Long uid = req.getUid();
+        BigDecimal amount = req.getAmount();
+        // make a deposit
+        User user = userService.getUserByID(uid);
+        boolean success = userService.makeADeposit(user, amount);
+        // payload
+        BalanceRsp res = new BalanceRsp();
+        res.setSuccess(success);
+        res.setCurrBalance(user.getBalance());
 
-    @PostMapping("/register")
-    public ResponseEntity<?> register(@Valid @RequestBody RegisterReq registerReq, Errors errors) {
-        if (userRepository.existsByEmail(registerReq.getEmail())) {
-            List<String> errorFields = new ArrayList<>();
-            errorFields.add("email");
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new ErrorRsp(errorFields, "Email already registered!"));
-        }
-        if (accountRepository.existsByNumberAndRoutingNumber(registerReq.getAccountNumber(), registerReq.getRoutingNumber())) {
-            List<String> errorFields = new ArrayList<>();
-            errorFields.add("accountNumber");
-            errorFields.add("rountingNumber");
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new ErrorRsp(errorFields, "Bank account already registered!"));
-        }
+        return success ? ResponseEntity.status(HttpStatus.OK).body(res) :
+                        ResponseEntity.status(HttpStatus.NOT_FOUND).body(new ErrorRsp(new ArrayList<>(Arrays.asList("i think we need to refactor this")), "Error in making a deposit."));
 
-        var user = new User();
-        user.setEmail(registerReq.getEmail());
-        user.setPassword(registerReq.getPassword());  // TODO spring security password encoding
-        user.setTypeByStr(registerReq.getUserType());
-        user.setPhone(registerReq.getPhone());
-        user.setAddress(registerReq.getAddress());
-
-        var account = new Account();
-        account.setName(registerReq.getAccountName());
-        account.setNumber(registerReq.getAccountNumber());
-        account.setRoutingNumber(registerReq.getRoutingNumber());
-        user.setAccount(account);
+    }
+    @PutMapping("/withdraw")
+    public ResponseEntity<?> withdraw(@Valid @RequestBody BalanceReq req) {
+        // get params
+        Long uid = req.getUid();
+        BigDecimal amount = req.getAmount();
+        // make a withdraw
+        User user = userService.getUserByID(uid);
+        boolean success = userService.makeAWithdraw(user, amount);
+        // payload
+        BalanceRsp res = new BalanceRsp();
```

Comment:
qq: should we have a service for handling responses?

Reply:
I think you can move the payload initialization to service, example: https://github.com/osopromadze/Spring-Boot-Blog-REST-API/blob/master/src/main/java/com/sopromadze/blogapi/service/TagService.java